src/node_type.rs

```diff
@@ -663,8 +659,22 @@ impl InternalNode {
         unreachable!("Impossible to get here without returning even at the lowest level.")
     }
 
+    fn test_get_only_child(width: u8, range_existence_bitmap: u16, range_leaf_bitmap: u16) -> bool {
```

Comment:
nit: the name `test_get_only_child` sounds like it's meant to be used for testing. How about `has_only_child`?

Also, I don't think this needs to be an associated function - just a private helper seems a little bit more natural.